libraries/AP_LocationDB/AP_LocationDB_config.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL_Boards.h>
+#include <AP_ADSB/AP_ADSB_config.h>
+#include <AP_Scripting/AP_Scripting_config.h>
+
+#ifndef AP_LOCATIONDB_ENABLED
+#define AP_LOCATIONDB_ENABLED BOARD_FLASH_SIZE > 1024
```

Comment:
This needs to be refined. Ideally it should be enabled based on the features that require it (OA at the moment?). Either way I think we should now avoid putting board size checks in the source code, but instead use whatever is the final result of @peterbarker 's minimal_config.inc

Reply:
> This needs to be refined. Ideally it should be enabled based on the features that require it (OA at the moment?). Either way I think we should now avoid putting board size checks in the source code, but instead use whatever is the final result of @peterbarker 's minimal_config.inc

I actually suggested this change - previously it was on on all boards, including 1MB boards.

We do have some 1MB boards which haven't been minimized yet, and I don't think we can use the minimize files.

I *have* wondered about how we might remove the BOARD_FLASH_SIZE checks throughout the code, but haven't really come up with any great ideas.  We could add them in the `chibios_hwdef.py` in-line includes, I guess?  In any case, I don't think the pattern I suggested here is uncommon in AP, so do you think we should hold things up here?

Reply:
Why not make it dependent on the features that are using it? I think these are fairly specific

Also shouldn't below be:
```
#define AP_LOCATIONDB_KEYDOMAIN_ADSB_ENABLED HAL_ADSB_ENABLED &&  AP_LOCATIONDB_ENABLED
```
For example, otherwise switching it off is hard

Reply:
Yep, the individual key domains would probably look something like this:

```
#define AP_LOCATIONDB_KEYDOMAIN_ADSB_ENABLED AP_LOCATIONDB_KEYDOMAIN_DEFAULT_ENABLED && HAL_ADSB_ENABLED
```

That's the way we're moving with a lot of our libraries.

We could potentially make it dependent on whether there's any consume of the data - could you investigate that, @shiv-tyagi , please?

Reply:
> Yep, the individual key domains would probably look something like this:

I have applied these changes. Thanks.



> We could potentially make it dependent on whether there's any consume of the data - could you investigate that, @shiv-tyagi , please?
Yes, we can probably make it dependent on MODE_FOLLOW_ENABLED and AC_AVOID_ENABLED.

Reply:
Update: I tried making AP_LOCATIONDB_ENABLED dependent on AP_FOLLOW_ENABLED (which is one of the consumers of the data) and it did not work out because AP_FOLLOW_ENABLED itself is dependent on AP_LOCATIONDB_ENABLED.